pkg/homedir/homedir_unix.go

```diff
@@ -101,9 +101,39 @@ func isWriteableOnlyByOwner(perm os.FileMode) bool {
 	return (perm & 0o722) == 0o700
 }
 
-// GetConfigHome returns XDG_CONFIG_HOME.
+// ConfigHome returns XDG_CONFIG_HOME.
+// ConfigHome returns $HOME/.config and nil error if XDG_CONFIG_HOME is not set.
+// Verifies ownership of config directory matches the current process or
+// returns error.
+// See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
+func GetConfigHome() (string, error) {
+	rootlessConfigHomeDirOnce.Do(func() {
+		tmpDir := os.Getenv("XDG_CONFIG_HOME")
```

Comment:
(`tmpDir` is a surprising name for this value to me in this context, I’d read it as imposing a “temporary directory” semantics.

I’d expect “configDir” or something like “candidate”.)